modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/LogicalOrToUnionRule.java

```diff
@@ -24,40 +24,31 @@
 import org.apache.calcite.plan.RelOptRuleCall;
 import org.apache.calcite.plan.RelOptUtil;
 import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelCollation;
 import org.apache.calcite.rel.RelNode;
 import org.apache.calcite.rel.core.RelFactories;
-import org.apache.calcite.rel.logical.LogicalFilter;
 import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexLocalRef;
 import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexShuttle;
 import org.apache.calcite.rex.RexUtil;
 import org.apache.calcite.sql.SqlKind;
 import org.apache.calcite.tools.RelBuilder;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.apache.calcite.util.mapping.Mappings;
 import org.apache.ignite.internal.processors.query.calcite.rel.logical.IgniteLogicalTableScan;
+import org.apache.ignite.internal.processors.query.calcite.schema.IgniteIndex;
+import org.apache.ignite.internal.processors.query.calcite.schema.IgniteTable;
+import org.apache.ignite.internal.processors.query.calcite.type.IgniteTypeFactory;
+import org.apache.ignite.internal.processors.query.calcite.util.Commons;
+import org.apache.ignite.internal.util.typedef.F;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * Converts OR to UNION ALL.
  */
 public abstract class LogicalOrToUnionRule extends RelRule<LogicalOrToUnionRule.Config> {
-    /** Rule instance to replace filters. */
-    public static final RelOptRule FILTER_INSTANCE = new LogicalOrToUnionRule(Config.FILTER) {
-        @Override protected RexNode getCondition(RelOptRuleCall call) {
-            final LogicalFilter rel = call.rel(0);
-
-            return rel.getCondition();
-        }
-
-        @Override protected RelNode getInput(RelOptRuleCall call) {
-            final LogicalFilter rel = call.rel(0);
-
-            return rel.getInput();
-        }
-
-        @Override protected void buildInput(RelBuilder relBldr, RelNode input, RexNode condition) {
-            relBldr.push(input).filter(condition);
-        }
-    };
-
     /** Rule instance to replace table scans with condition. */
     public static final RelOptRule SCAN_INSTANCE = new LogicalOrToUnionRule(Config.SCAN) {
```

Comment:
It's only one instance now, let's rename it to `INSTANCE` and made related changes to the class (remove abstract class, move logic from anonymous class to the parent class, since it's already code where `IgniteLogicalTableScan` is assumed in the abstract class)